internal/entities/handlers/message/message.go

```diff
@@ -0,0 +1,98 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package message contains the message creation strategies
+package message
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+
+	"github.com/stacklok/minder/internal/entities/properties"
+	v1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+// TypedProps is a struct that contains the type of entity and its properties.
+// it is used for either the entity or the owner entity.
+type TypedProps struct {
+	Type       v1.Entity      `json:"type"`
+	GetByProps map[string]any `json:"get_by_props"`
+}
+
+// EntityHint is a hint that is used to help the entity handler find the entity.
+type EntityHint struct {
+	ProviderImplementsHint string `json:"provider_implements_hint"`
+}
+
+// HandleEntityAndDoMessage is a message that is sent to the entity handler to refresh an entity and perform an action.
+type HandleEntityAndDoMessage struct {
+	Entity TypedProps `json:"entity"`
+	Owner  TypedProps `json:"owner"`
```

Comment:
Owner can be a little confusing (at first I thought we were conflating GitHub terms again). Can we rename this to originator or begetter?